core/src/main/scala/query/model.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package freestyle.cassandra
+package query
+
+import java.nio.ByteBuffer
+
+import cats.MonadError
+
+object model {
+
+  trait SerializableValue {
+    def serialize[M[_]](implicit E: MonadError[M, Throwable]): M[ByteBuffer]
```

Comment:
Since users may have to provide instances for this at some point, does it make sense that `serialize` is defined as? :
```scala
def serialize: Either[Throwable, ByteBuffer]
```
We can always lift `Either[Throwable, ByteBuffer]` ourselves with `ApplicativeError[M, Throwable]#fromEither`

Reply:
You mean to revert #74 or import and use Try [here](https://github.com/frees-io/freestyle-cassandra/pull/80/files#diff-34175872c3aa409b815ac1e9a03f9648R47)

Reply:
No, it's fine, I thought we could simplify the public API but is not a big deal.